helix-core/src/main/java/org/apache/helix/monitoring/mbeans/JobMonitor.java

```diff
@@ -182,6 +198,59 @@ public void updateControllerInducedDelayGauge(long delay) {
     _controllerInducedDelayGauge.updateValue(delay);
   }
 
+  /**
+   * Sync the current SubmissionToProcessDelay mean to ZooKeeper
+   * @param baseDataAccessor
+   */
+  public void syncSubmissionToProcessDelayToZk(BaseDataAccessor<ZNRecord> baseDataAccessor) {
```

Comment:
As discussed offline, this implementation could be improved.
- This is too specific and not very easy to maintain or scalable - suppose you want to add more metrics that you need to persist, then you'd have to implement methods for each additional metric.
- Can you look into inversion of control? Perhaps you could create a component with the appropriate set of interface methods that allow persisting of metric numbers to a metadata store in a more generic fashion. For example, you could have something like,

> <I> MetricStorage
> <Class> ZkMetricStorage (takes a Zk connection to initialize)

Also, this MetricStorage implementation could take in a dynamic list of metric names to be persisted, etc. You could get creative with this. No need to feel like you have to cover all use cases, but designing a component in a way that is extendable and generic enough will save you and help with maintainability.